lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -287,6 +358,175 @@ FailureOr<BinaryFlags> getBinaryFlags(Type operandType, Type outputType,
   return failure();
 }
 
+FailureOr<int64_t> getLeadingDim(Type type, size_t pos) {
+  // Not shaped type, the leading dimension is the single scalar.
+  auto memref = dyn_cast<MemRefType>(type);
+  if (!memref)
+    return 1;
+  // For 1d memref we cannot use the stride as leading dimension, but the
+  // leading dimension is the dimension itself.
+  if (memref.getRank() == 1)
+    return memref.getShape()[0];
+
+  SmallVector<int64_t> strides;
+  int64_t offset;
+  if (failed(getStridesAndOffset(memref, strides, offset)))
+    return failure();
+  // fail if the strides are non-constant
+  if (llvm::any_of(strides, [](int64_t stride) {
+        return stride == ShapedType::kDynamic;
+      }))
+    return failure();
+  return strides[pos];
+}
+
+bool isTwoDTransposeOp(vector::TransposeOp transposeOp) {
```

Comment:
Sounds like a simpler approach would be to create a function to get the rank of a vector (in view of VNNI layout, which just needs the operand).

Reply:
This funciton belongs in the brgemm patch. Removing this.

Reply:
you still seem to be using in the code? Is this really just for the brgemm patch?

Reply:
Oh i thought you were referring to the line above it, you were referring to the twoDtransposeOp. I will update it to use ranks.